ethereum_history_api/circuits/lib/src/misc/bytes.nr

```diff
@@ -0,0 +1,53 @@
+use dep::proof::utils::byte_value;
+
+pub fn left_nibble_shift<N>(nibbles: [u8; N]) -> ([u8; N], u64) {
+    let (mut shifted, mut len) = byte_value(nibbles);
+    if (len % 2 == 1) {
+        right_shift_by_one(&mut shifted);
+        len += 1;
+    }
+
+    let shifted_by_len = N - len;
+    (shifted, shifted_by_len)
+}
+
+fn right_shift_by_one<N>(arr: &mut [u8; N]) {
+    let mut last = 0;
+    for i in 0..N {
+        let tmp = (*arr)[i];
+        (*arr)[i] = last;
+        last = tmp;
+    }
```

Comment:
Move to `nibbles_to_bytes`

Reply:
I'm not sure if it's a good idea to move `shift_right_by_one` to `nibbles_to_bytes`. I think right now `left_to_right_padding` has responsibility of moving the nibbles within the array and `shift_right_by_one` is part of moving nibbles within the array. Also, `nibbles_to_bytes` has the responsibility to convert given nibbles to bytes and if we move `shift_right_by_one` here it won't be that clear what `nibbles_to_bytes` does.

Reply:
Yes, you convinced me :)